torch/_subclasses/functional_tensor.py

```diff
@@ -501,7 +501,24 @@ def inner(*args, **kwargs):
         disable_above = torch._C._ExcludeDispatchKeyGuard(
             torch._C.DispatchKeySet(torch._C.DispatchKey.Functionalize)
         )
-        with disable_above, mode:
+        # FIXME (tmanlaibaatar) This is little weird.
+        # when map_impl.autograd is invoked, we shouldn't be
+        # tracing through predispach mode stack. But since this mode is
+        # saved before this happens on the PythonFunctinalizeAPI object,
+        # it will still try to dispatch through PreDispach FunctionalTensorMode
```

Comment:
I don't think I follow - if pre-dispatch tracing is turned on, then what I think I would expect is:

(1) we hit the PreDispatch key and run our functional_mode / proxy_mode first, before autograd runs

(2) when functional_mode/proxy_mode, runs, we pop it off the stack and redispatch

(3) the autograd impl for map() runs next, but at this point proxy/functional modes have finished running, and should not get redispatched.

Does that not line up with what you're seeing?